src/ui/components/search/searchcomponent.js

```diff
@@ -150,11 +160,22 @@ export default class SearchComponent extends Component {
       }
       if (q === null) {
         if (this._defaultInitialSearch || this._defaultInitialSearch === '') {
+          this.core.globalStorage.set(StorageKeys.QUERY_TRIGGER, QueryTriggers.INITIALIZE);
           this.core.setQuery(this._defaultInitialSearch);
         }
         return;
       }
-      this.debouncedSearch(q);
+
+      const queryTrigger = this.core.globalStorage.getState(StorageKeys.QUERY_TRIGGER);
+      const resetPagination = this._verticalKey &&
+        queryTrigger !== QueryTriggers.QUERY_PARAMETER &&
+        queryTrigger !== QueryTriggers.INITIALIZE;
```

Comment:
Do we anticipate making a `QueryTrigger` for manual input in the search bar? If not, could we simplify this logic by saying if `verticalKey` and if `!queryTrigger`, then `resetPagination`?

Reply:
I was not sure what to anticipate! We currently have another query trigger (for SpellCheck component, which is more dynamic, relying on the `type` of the data returned). I believe we would want spellcheck query triggers to continue to reset pagination. It felt easier here to enumerate these two than try to figure out the possible types for SpellCheck. How does that sound to you?